packages/horizon/contracts/data-service/libraries/DataServiceFeesLib.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.27;
+
+import { ProvisionTracker } from "./ProvisionTracker.sol";
+import { IDataServiceFees } from "../interfaces/IDataServiceFees.sol";
```

Comment:
I would remove the dependency with `IDataServiceFees` any event or error that we use from there I think should probably instead live in this lib contract. Note that if we move any errors into this file we would need to rename their prefix `IDataServiceFees.DataServiceFeesClaimNotFound` -> `StakeClaimsClaimNotFound`

Reply:
ah yes that makes sense. to avoid name collision we can rename the struct to `State` similar to what we have with `Allocation.State` and a few others.